docs/design/dfx-extensions.md

```diff
@@ -0,0 +1,65 @@
+
+## dfx exension install
+
+The dfx utility offers a feature to install new extensions. Here's a high-level overview of how the installation process works:
+
+1. Determine Extension Compatibility:
+    - Before any installation can take place, dfx checks if the requested extension is already installed. If it is, the process terminates with an error.
+    - The utility identifies the version of the extension compatible with its own version. This ensures that users don't end up installing extensions that may not work properly with their specific version of dfx.
+    - Compatibility is determined using an extension compatibility matrix, which presumably maps extensions to compatible versions of dfx.
+    - The compatibility matrix is a map that relates dfx versions to a set of extensions and their versions.
+    - This matrix is fetched from a predefined URL. This means that dfx can always retrieve the most up-to-date compatibility information by querying this URL.
+    - When checking for compatibility, dfx consults this matrix to determine the latest version of an extension that's compatible with the current dfx version.
+    - This approach ensures that dfx only runs or interacts with extensions that are guaranteed to work correctly with its current version, preventing potential mismatches or runtime errors.
+    - Given a dfx version and an extension name, the system checks the compatibility matrix.
+    - If the given dfx version is not in the matrix, an error is raised.
+    - If the extension name is not associated with the dfx version, another error is raised.
+    - The versions of the extension that are compatible with the dfx version are then sorted, and the latest version is chosen as the most suitable one.
+    - This mechanism ensures that users always get the newest version of an extension that works correctly with their dfx version.
+    - In essence, while dfx provides tools to manage extensions, the compatibility matrix and extension manifest add an extra layer of sophistication. They ensure that extensions are accurately described, can be transformed into command-line interfaces, and most importantly, are always in sync with the version of dfx the user is operating. This comprehensive management system guarantees smooth operations and enhances user experience.
+
+2. Download and Extraction:
+    - Once the compatible version of the extension is determined, dfx constructs a download URL. This URL points to a GitHub releases page where the extensions are hosted. Currently, extensions support downloading only from `dfinity/dfx-extensions` repository. Here is how the URL template looks like:
+    ```
+    "https://github.com/dfinity/dfx-extensions/releases/download/{github_release_tag}/{extension_archive_name}.tar.gz";
+    ```
+    - The extension is then downloaded from this URL. If the download fails for any reason, an error is returned.
```

Comment:
Should be no indentation for the triple-backquote
Should be a blank line after

Reply:
github diff-view seems to incorrectly render it
see rendered view, looks alright there https://github.com/dfinity/sdk/blob/caefa3588ac04446a0c818a9c431d7da2a129673/docs/design/dfx-extensions.md